### PR TITLE
Multi topic filter

### PR DESCRIPTION
**Pull request by mmazanec22**
Adds multi topic filter.  Pulls data fetching into a context.  Reorgs some things into utils.

Note: asking for PR review from a few people not because everyone actually needs to review this, but because I'm not sure who's busy/wants to know what's up/etc.

### CODE REVIEW

**Comment by nihonjinrxs** [comment]
I'll try to take a closer look at the code sometime tomorrow if others don't get to it before me. Looking at the deploy preview, though, it seems pretty functional.

One thought -- since projects may match one or many tags, it might be nice to highlight the matching tags in the project records so it's easy to see how many of the selected tags match. Also might be nice to resort the projects so that ones matching more tags show up higher in the list.

**Review by tdooner — APPROVED** [review]
This is well above my current level of experience with React, so I mostly just read through it and nodded, appreciating the experience to stay up-to-date on the code structure.

As a result, I don't have any architectural feedback to give. I'd say to feel free to merge!

File: statusboard/src/utils.js [hunk 1]
```diff
@@ -48,13 +48,53 @@ export function getProjectsFromBrigadeData(brigadeData) {
...
+    topics: [],
+    ...options,
+  };
+
+  return projects
+    .map((p) => ({
+      ...p,
+      numberTopicsMatched: topicsIntersect(topics, p.topics),
+    }))
+    .sort((a, b) => b.numberTopicsMatched - a.numberTopicsMatched)
```
[the remark this answers is not in the document]
Comment (mmazanec22) [reply statusboard/src/utils.js:73]: We could implement that by adding an and/or toggle.  I'll open a ticket!

File: statusboard/src/pages/Projects/Projects.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      `?${stringify(
+        {
+          tags: filterTopics || tags,
+          timeRange: activeThreshold || timeRange,
+        },
+        { arrayFormat: 'comma' }
+      )}`
+    );
+    // eslint-disable-next-line
+  }, [history, allProjects, activeThreshold, filterTopics]);
```
[the remark this answers is not in the document]
Comment (mmazanec22) [reply statusboard/src/pages/Projects/Projects.js:70]: Yeah I sort of hate that react makes us include all sorts of crap in the deps, especially when it's obviously a bad idea to rerender based on a certain value (like one that you're setting in that hook).  Really hoping they sort that out in some future release.

**Pull request merged**